pkg/graph/resolvers/system_profile_section_locks.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	delete(systemProfileSessionLocks.systemLockStatuses[cedarSystemID], section)
+
+	// Clean up empty system map
+	if len(systemProfileSessionLocks.systemLockStatuses[cedarSystemID]) == 0 {
+		delete(systemProfileSessionLocks.systemLockStatuses, cedarSystemID)
+	}
+
+	systemProfileSessionLocks.Unlock()
+
+	// Publish outside the lock to avoid blocking other operations
+	sessionID := cedarSystemIDToSessionID(cedarSystemID)
+	ps.Publish(sessionID, pubsubevents.SystemProfileSectionLocksChanged, models.SystemProfileSectionLockStatusChanged{
+		ChangeType: models.LockChangeTypeRemoved,
+		LockStatus: &lockStatus,
+		ActionType: actionType,
+	})
+
+	return true, nil
+}
+
+// isUserAuthorizedToEditLock checks if a user is authorized to unlock a section.
+// Users can only unlock sections they personally locked.
+func isUserAuthorizedToEditLock(lockStatus models.SystemProfileSectionLockStatus, userID uuid.UUID) bool {
+	return userID == lockStatus.LockedByUserAccount.ID
+}
+
+// UnlockAllSystemProfileSections is an admin function that unlocks all sections for a system.
+// Bypasses ownership checks - can unlock sections owned by any user.
+// Publishes REMOVED events with ADMIN action type for each unlocked section.
+func UnlockAllSystemProfileSections(ps pubsub.PubSub, cedarSystemID string) ([]*models.SystemProfileSectionLockStatus, error) {
+	systemProfileSessionLocks.Lock()
+
+	sectionLocks, found := systemProfileSessionLocks.systemLockStatuses[cedarSystemID]
+	if !found {
+		systemProfileSessionLocks.Unlock()
+		return nil, nil
+	}
+
+	// Copy sections to unlock while holding the lock
+	var sectionsToUnlock []struct {
+		section    models.SystemProfileLockableSection
+		lockStatus models.SystemProfileSectionLockStatus
+	}
+
+	for section, lockStatus := range sectionLocks {
+		sectionsToUnlock = append(sectionsToUnlock, struct {
+			section    models.SystemProfileLockableSection
+			lockStatus models.SystemProfileSectionLockStatus
+		}{section: section, lockStatus: lockStatus})
+	}
+
+	// Delete from map while holding lock
+	delete(systemProfileSessionLocks.systemLockStatuses, cedarSystemID)
+	systemProfileSessionLocks.Unlock()
+
+	// Publish events outside the lock
+	sessionID := cedarSystemIDToSessionID(cedarSystemID)
+	var unlockedSections []*models.SystemProfileSectionLockStatus
+	for _, item := range sectionsToUnlock {
+		unlockedSections = append(unlockedSections, &item.lockStatus)
+		ps.Publish(sessionID, pubsubevents.SystemProfileSectionLocksChanged, models.SystemProfileSectionLockStatusChanged{
+			ChangeType: models.LockChangeTypeRemoved,
+			LockStatus: &item.lockStatus,
+			ActionType: models.LockActionTypeAdmin,
+		})
+	}
+
+	return unlockedSections, nil
+}
+
+// getOwnedSections returns a list of system profile sections owned by a specific principal
+func getOwnedSections(sectionLocks sectionLockStatusMap, subscriber pubsub.Subscriber) []models.SystemProfileLockableSection {
```

Comment:
Thanks! I don't think we need to do this now, but I think we could potentially use pointers stored in the map, and have multiple maps.

So instead of iterating through all sections to see if someone has locked it, we just have one map to get the Section by section enum, and then another map that stores the sections by owner id. Since it is a pointer, you can update either and the data is updated. It's a bit of a departure from the MINT implementation, and I don't think it's a huge deal to continue with the current paradigm. But it's probably an opportunity for improvement.